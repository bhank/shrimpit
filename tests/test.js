var Shrimpit = require("../shrimpit");

test("should be able to instantiate Shrimpit", () => {
    expect(new Shrimpit()).toBeDefined();
});

test("test directory a", () => {
    const shrimpit = new Shrimpit();
    shrimpit.exec("./test/a");
    expect(shrimpit.modules.exports).toMatchSnapshot("a exports");
    expect(shrimpit.modules.imports).toMatchSnapshot("a imports");
    expect(shrimpit.getUnresolved()).toMatchSnapshot("a unresolved");
});

test("test directory b", () => {
    const shrimpit = new Shrimpit();
    shrimpit.exec("./test/b");
    expect(shrimpit.modules.exports).toMatchSnapshot("b exports");
    expect(shrimpit.modules.imports).toMatchSnapshot("b imports");
    expect(shrimpit.getUnresolved()).toMatchSnapshot("b unresolved");
});

describe("deExtensionize", () => {
    const shrimpit = new Shrimpit();
    [
        {path: "./dir/subdir/file.ext", expected: "./dir/subdir/file"},
        {path: "./dir/subdir/index.js", expected: "./dir/subdir"},
        {path: "./dir/subdir/", expected: "./dir/subdir"},
        {path: ".\\dir\\subdir\\file.ext", expected: ".\\dir\\subdir\\file"},
        {path: ".\\dir\\subdir\\index.js", expected: ".\\dir\\subdir"},
        {path: ".\\dir\\subdir\\", expected: ".\\dir\\subdir"},
    ].forEach(data => {
        test(`should deExtensionize '${data.path}'`, () => expect(shrimpit.deExtensionize(data.path)).toEqual(data.expected));
    });
});