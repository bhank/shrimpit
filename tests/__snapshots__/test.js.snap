// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a exports 1`] = `
Array [
  "a11",
  "a12",
  "a2",
  "a4",
  "test/a/a4",
]
`;

exports[`a imports 1`] = `
Array [
  "test/a/a4",
  "test",
  "a11",
  "a12",
]
`;

exports[`a unresolved 1`] = `
Set {
  "a2",
  "a4",
}
`;

exports[`b exports 1`] = `
Array [
  "b1",
  "test/b/b1",
  "test/b/b2",
]
`;

exports[`b imports 1`] = `Array []`;

exports[`b unresolved 1`] = `
Set {
  "b1",
  "test/b/b1",
  "test/b/b2",
}
`;
